### ISSUE TITLE
"Expected 1 args but only found 0" is too vague

### ISSUE THREAD

**Issue by brentp** [iffy/nim-argparse#18]
this code shows 2 behaviors that I hope might be improved:
1. If an option (`--info` below) is specified multiple times by the user, no warning is given and only the last is used. I would prefer this to give an error or at least a warning
2. If `parse()` is called without arguments, then it gives an exception that's not helpful to the user.
Is there something I can do to make outputting the help as a default? I always check if args.len == 0 and if so I set `args = @["--help"]`. 
thanks for the great library. I'm using it in all of my projects now.

```Nim
import argparse

var p = newParser("example"):
  option("--info")
  arg("arg")

var opts = p.parse(@["--info", "asdf", "--info", "asdf", "arg"])
echo opts

discard p.parse()
```

**Comment by iffy** [comment]
Thanks for reporting this!

1. I've filed issue #19 for that first item.
2. If you use `p.run()` instead of `p.parse()` it might do what you're expecting.  I think I intended the `run` path to be automatically user-friendly and the `parse` path to rely on the program to format nice messages for users.  (I'm on a fresh computer without Nim installed at the moment, so I can't test yet).  But I'd be open for improvements here.
3. Same for this.  If I remember right, `p.parse()` with no args will have `opts.help` set or not depending on if `--help` was used on the command line.  But if you use `p.run()` with no args, it will display the help text or not depending on the command line.

**Comment by brentp** [comment]
I actually didn't know about `p.run`. I see it's mentioned in the docs (after p.parse example) but might be good to have it prevalent in the readme.

**Comment by brentp** [comment]
the other thing I do in all cases is this:
```
var args = commandLineParams()
if args.len == 0: args = @["--help"]
var opts = p.parse(args)
if p.help:
    quit 0
```
because if the user runs the program without arguments, `p.parse()` will give a non informative message. And if help is set, I always want to exit with 0. I tried p.run() and it doesn't seem to help in those cases either.

**Comment by iffy** [comment]
@brentp (I finally have Nim set up on this machine again)

Okay, here's the error it's throwing.  I see what you mean about it not being helpful to users:

```
/private/tmp/goo.nim(10) goo
/Users/matt/m/nim-argparse/src/argparse.nim(657) parse
/Users/matt/m/nim-argparse/src/argparse.nim(651) parse
/Users/matt/m/nim-argparse/src/argparse.nim(579) parse
Error: unhandled exception: Expected 1 args but only found 0 [UsageError]
```

It would be better if it said something like:

`Expected 1 value for 'arg' argument.  Run with --help for more information.`

I don't think I can make the default no-argument behavior to display the help text because a lot of programs expect to run without arguments (liked `ls` or `cat` for instance).

Maybe this is an okay solution, though?

```nim
import argparse

var p = newParser("example"):
  option("--info")
  arg("arg")

try:
  discard p.parse()
except UsageError:
  echo getCurrentExceptionMsg()
  echo p.help
```

**Comment by brentp** [comment]
thanks for the reply (and fix for repeated arguments).

I think your reply for no args should be ok. but if we have e.g. a single `arg` with `nargs` which is not `-1` then we can know an argument is required.

**Comment by eugenesvk** [comment]
> Maybe this is an okay solution, though?

How would this solution help me if I want to use `let opts = p.parse()` for further parsing the command line options? If I use it after the try/except, I get duplicate help message when `-h` flag is passed (so discard doesn't discards the help message output?), but if I use it instead of `discard p.parse()` then the opt scope is limited to the block


> I don't think I can make the default no-argument behavior to display the help text

Can you maybe make it an additional option for positional arguments, e.g. `arg("arg", helpOnError)`, which would be defined as `proc arg(...helpOnError: bool = false...) {...}` or something similar?

**Comment by iffy** [comment]
Fixed in v1.0.0
